### PR TITLE
feat: LSDV-4830: Add option to provide image dimensions in YOLO import so we don't have to open image

### PR DESCRIPTION
**Pull request by Axel-Jacobsen**
When converting a large number of labels into a tasks.json, especially when the data is mounted via a remote drive, the image opening process can be painfully slow. There are times where

- all images in the dataset are the same size, and
- we need to recreate `tasks.json` frequently

I made these changes to a custom branch of `label-studio-converter`, but I figured that y'all may like the changes too.

Also, when will the next release be? The YOGO image path fix would be great for us, we are currently just cloning your repo to get the most up-to-date bugfixes.

Note: your contributing docs suggested formatting w/ `black --skip-string-normalization`, so that is why there are many more diffs than one would expect for this.

### CODE REVIEW

File: label_studio_converter/imports/yolo.py [hunk 1]
```diff
@@ -117,12 +142,14 @@ def convert_yolo_to_ls(input_dir, out_file,
...
-              f'     https://labelstud.io/guide/storage.html#Local-storage]\n'
-              f'  4. Import "{out_file}" to the project\n')
+        print(
+            '\n'
+            f'  1. Create a new project in Label Studio\n'
+            f'  2. Use Labeling Config from "{label_config_file}"\n'
+            f'  3. Setup serving for images [e.g. you can use Local Storage (or others):\n'
+            f'     https://labelstud.io/guide/storage.html#Local-storage]\n'
+            f'  4. Import "{out_file}" to the project\n'
+        )
```
Comment (Axel-Jacobsen) [inline label_studio_converter/imports/yolo.py:152]: This should be logging, no?

File: label_studio_converter/imports/yolo.py [hunk 2]
```diff
@@ -1,33 +1,44 @@
 import os
```
Comment (Axel-Jacobsen) [inline label_studio_converter/imports/yolo.py:1]: I think that `pathlib` could simplify some code below, quite nicely

**Review by Axel-Jacobsen — COMMENTED** [review]
just a couple of quick comments on future improvements that I may make, curious on your thoughts

File: label_studio_converter/imports/yolo.py [hunk 3]
```diff
@@ -131,38 +158,60 @@ def add_parser(subparsers):
...
         default='.jpg',
     )
+    yolo.add_argument(
+        '--image-dims',
+        dest='image_dims',
+        help=(
+            "optional tuple of integers specifying the image width and height of *all* "
+            "images in the dataset. Defaults to opening the image to determine it's width "
+            "and height, which is slower. This should only be used in the special "
+            "case where you dataset has uniform image dimesions. e.g. `--image-dims 600 800` "
```
Comment (makseq) [inline label_studio_converter/imports/yolo.py:213]: Not sure it's working - `--image-dims 600 800` - because 600 800 has a space in between..

Comment (Axel-Jacobsen) [reply label_studio_converter/imports/yolo.py:215]: Good catch, sorry for the delay! Tests incoming

File: label_studio_converter/imports/yolo.py (hunk 1 above)
Comment (makseq) [reply label_studio_converter/imports/yolo.py:152]: yes, we can do it as logging, but we should be sure that users will see it 100%

File: label_studio_converter/imports/yolo.py (hunk 3 above)
Comment (makseq) [reply label_studio_converter/imports/yolo.py:215]: @Axel-Jacobsen Do you have any news about tests? Will you be able to add them?

Comment (Axel-Jacobsen) [reply label_studio_converter/imports/yolo.py:215]: Ah sorry! It has been hectic at work, I'll work on these this week pinky promise

Comment (makseq) [reply label_studio_converter/imports/yolo.py:216]: yes, awesome!

**Comment by makseq** [comment]
@Axel-Jacobsen Do we have issue dedicated to this PR? If yes, please mention it here.

**Pull request merged**